pensieve/external_config.py

```diff
@@ -46,7 +47,15 @@ def from_github_repo(cls):
         for file in files:
             if file.name.endswith(".toml"):
                 normandy_slug = os.path.splitext(file.name)[0]
-                configs.append(ExternalConfig(normandy_slug, file.decoded_content, file.last_modified))
+                spec = AnalysisSpec.from_dict(toml.loads(file.decoded_content))
+                configs.append(ExternalConfig(normandy_slug, spec, file.last_modified))
 
         return cls(configs)
 
+    def spec_for_experiment(self, normandy_slug):
```

Comment:
ExperimentCollection filter methods always return a (potentially empty) ExperimentCollection; it'd be nice to be consistent.

Reply:
Oh I was going to delete this comment but I forgot; I think your PR is reasonable here.

Although: one advantage we have here compared to ExperimentCollection is that we know we're going to refer to these by slug and we know the slugs have to be unique, so we could wrap a dictionary here instead of wrapping a list that we iterate over -- but I don't think it's consequential.